### PR TITLE
mqtt.cpp libraries

### PR DESCRIPTION
**Pull request by praveen-garg**

### CODE REVIEW

File: others/scripts/mqtt/requirements.sh [hunk 1]
```diff
@@ -53,6 +53,18 @@ git clone https://github.com/eclipse/paho.mqtt.c.git
 cd paho.mqtt.c
 mkdir build
 cd build
-cmake -DPAHO_BUILD_DOCUMENTATION=FALSE -DPAHO_WITH_SSL=FALSE ..
+cmake -DPAHO_BUILD_DOCUMENTATION=FALSE -DPAHO_WITH_SSL=TRUE ..
```
Comment (praveen-garg) [inline others/scripts/mqtt/requirements.sh:56]: I wonder how MQTT Scripted plugins are working with TLS/SSL?

File: others/scripts/mqtt/requirements.sh [hunk 2]
```diff
@@ -53,6 +53,18 @@ git clone https://github.com/eclipse/paho.mqtt.c.git
...
 make
 sudo make install
+
+cd ..
+cd ..
+
+rm -rf paho.mqtt.cpp
+git clone https://github.com/eclipse/paho.mqtt.cpp
+cd paho.mqtt.cpp
+cmake -Bbuild -H. -DPAHO_BUILD_STATIC=ON -DPAHO_WITH_SSL=ON -DPAHO_BUILD_DOCUMENTATION=FALSE -DPAHO_BUILD_SAMPLES=FALSE
```
Comment (praveen-garg) [inline others/scripts/mqtt/requirements.sh:66]: @gnandan Do we really need PAHO_BUILD_STATIC=ON ?

**Review by ashish-jabble — APPROVED** [review]
So we require new binaries for mqtt and packaged them into single common one's which is fine but make sure their with same binaries name for different repos will not be overwritten.

We may add new checkout dirs into the `.gitignore` file

**Review by gnandan — APPROVED** [review]

File: others/scripts/mqtt/requirements.sh (hunk 2 above)
Comment (gnandan) [reply others/scripts/mqtt/requirements.sh:66]: No, it's not required

**Pull request merged**